### ISSUE TITLE
cotainer_image run binary on build

### ISSUE THREAD

**Issue by Globegitter** [bazelbuild/rules_docker#285]
I am trying to create a docker container that includes bazel to run in our ci systems and it works all well, I depend on the deb package via `http_file` and add `openjdk-8-jdk-headless` as an additional package to the already provided java image, so my rules looks as follows:

```
container_image(
    name = "bazel_image",
    base = "@java_debug_image_base//image",
    workdir = "/home",
    symlinks = {
        "/usr/bin/javac": "/usr/lib/jvm/java-8-openjdk-amd64/bin/javac",
        "/usr/bin/jar": "/usr/lib/jvm/java-8-openjdk-amd64/bin/jar",
    },
    debs = [
        "@bazel_deb//file",               
        packages["openjdk-8-jdk-headless"],        
    ],
)
```

The only problem then when running `bazel-real` for the first time (or `bazel` if I was to add bash) it runs the extraction command which can easily take 10s. So having to wait this time each time on of our ci workers runs the image for the first time seems to be adding a lot of overhead. Not sure what the best way is to add something like this or if you feel this is in the scope of the container images, but imo it would be nice. One of the simplest ways I can imagine would be `execute_binary`. But that might also just add weight to the rule for a very small use-case.

The other option I think that would be nice if I could just drop in the `bazel-x.y.z-installer-linux-x86_64.sh` file instead of going the `deb` route, but I guess that would equally need something like `execute_binary`. 

Or is there another way this is possible I have not realised yet?

**Comment by mattmoor** [comment]
tl;dr I'm not aware of a good way to do this with `container_image`

I know [folks](https://github.com/GoogleCloudPlatform/cloud-builders/blob/master/bazel/Dockerfile#L38) that have built such containers with `Dockerfile` will `RUN bazel version` to cache the expanded step, but this is a hack :(

It's possible that a proper Bazel install does this expansion via post-install hooks (which we don't support), but it'd be nice if the `deb` just contained the pre-extracted install.

I think your best bet will be to ask for better `deb` packaging on the main repo, since IDK if/what we can do here (without implementing things we've very intentionally avoided, e.g. `RUN`).
